transhumandesign/kag-base
# [fixed] Doors shaking when loading maps

## Status

- **READY**: this PR is (to the best of your knowledge) ready to be incorporated into the game.

## Description

Fixes #1337.

This change makes it so doors that are loaded with the map won't shake from not being adjacent to walls.

During testing, I saw that the code that checks if a door should start shaking is run on game times between 2 and 30.
I tested on `Princess04` and loaded that map over and over. The two door tiles that are not adjacent to walls would print game time 2, 3 then 3, 4, then 4, 5 etc. until 30. I'm not sure why it is like this. But since it never went above 30, I concluded that `return getGameTime() > 30 && !surface;` would be sufficient for fixing the issue. That means doors will only shake if not adjacent to walls and if the game time is greater than 30 ticks.

## Steps to Test or Reproduce

`/loadmap Princess04` or `/loadmap PrincessChallenge`.
Notice the two door tiles that are not adjacent to walls will shake. **After this PR, they won't.**
Build a wall of doors.
Notice they will shake. **After this PR, they will still shake.**
